### PR TITLE
Fixes #3336: Add ConverstionReview to `kubernetes-model-apiextensions`

### LINKED ISSUE #3336: Add ConversionReview to Kubernetes ApiExtensions Model
ConversionReview, ConversionRequest and ConversionResponse might be used by users who are interested in writing a conversion webhook in order to handle CustomResource conversions from one version to other.

You can find equivalent go struct here: https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/v1/types.go#L457

### PR DESCRIPTION
**Pull request by MUzairS15**
## Description

Fix #3336 
Added ConversionReview in APIExtensions Model 
## Type of change

 - [X] Bug fix (non-breaking change which fixes an issue)
 - [X] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [X] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [X] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [X] I tested my code in Kubernetes
 - [X] I tested my code in OpenShift

### CODE REVIEW

**Comment by MUzairS15** [comment]
@rohanKanojia Pls review #3363

File: kubernetes-model-generator/kubernetes-model-apiextensions/src/test/resources/valid-ConversionReview.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (rohanKanojia) [inline kubernetes-model-generator/kubernetes-model-apiextensions/src/test/resources/valid-ConversionReview.json:1]: Could you please rename the file to `valid-conversionreview.json`(all smallcases) just like other files? 

Please fix the indentation in this file too.

File: kubernetes-model-generator/kubernetes-model-apiextensions/src/test/java/io/fabric8/kubernetes/api/model/apiextensions/v1/ConversionReviewTest.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.fabric8.kubernetes.model.util.Helper;
+import org.junit.jupiter.api.Test;
+
+import static net.javacrumbs.jsonunit.core.Option.IGNORING_ARRAY_ORDER;
+import static net.javacrumbs.jsonunit.core.Option.IGNORING_EXTRA_FIELDS;
+import static net.javacrumbs.jsonunit.core.Option.TREATING_NULL_AS_ABSENT;
+import static net.javacrumbs.jsonunit.fluent.JsonFluentAssert.assertThatJson;
+
+public class ConversionReviewTest {
```
Comment (rohanKanojia) [inline kubernetes-model-generator/kubernetes-model-apiextensions/src/test/java/io/fabric8/kubernetes/api/model/apiextensions/v1/ConversionReviewTest.java:27]:
```suggestion
class ConversionReviewTest {
```

File: kubernetes-model-generator/kubernetes-model-apiextensions/src/test/java/io/fabric8/kubernetes/api/model/apiextensions/v1/ConversionReviewTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  @Test
+  public void conversionReviewTest() throws Exception {
```
Comment (rohanKanojia) [inline kubernetes-model-generator/kubernetes-model-apiextensions/src/test/java/io/fabric8/kubernetes/api/model/apiextensions/v1/ConversionReviewTest.java:32]:
```suggestion
  void testDeserializationAndSerialization() throws Exception {
```

**Review by rohanKanojia — CHANGES_REQUESTED** [review]
Looks good with minor comments :+1: 

Please add an entry to CHANGELOG.md regarding this change as well.

File: kubernetes-model-generator/kubernetes-model-apiextensions/src/test/resources/valid-ConversionReview.json (hunk 1 above)
Comment (MUzairS15) [reply kubernetes-model-generator/kubernetes-model-apiextensions/src/test/resources/valid-ConversionReview.json]: 👍

File: kubernetes-model-generator/kubernetes-model-apiextensions/src/test/java/io/fabric8/kubernetes/api/model/apiextensions/v1/ConversionReviewTest.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import static net.javacrumbs.jsonunit.fluent.JsonFluentAssert.assertThatJson;
+
+class ConversionReviewTest {
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  @Test
+  void testDeserializationAndSerialization() throws Exception {
+    // Given
+    final String originalJson = Helper.loadJson("/valid-ConversionReview.json");
```
Comment (rohanKanojia) [inline kubernetes-model-generator/kubernetes-model-apiextensions/src/test/java/io/fabric8/kubernetes/api/model/apiextensions/v1/ConversionReviewTest.java:34]:
```suggestion
    final String originalJson = Helper.loadJson("/valid-conversionreview.json");
```

**Comment by rohanKanojia** [comment]
ConversionReviewTest.java seems to be failing, Did you run the test before updating PR?

**Comment by MUzairS15** [comment]
> ConversionReviewTest.java seems to be failing, Did you run the test before updating PR?

Sorry missed the change in test, No I didn't as I just updated changelog.

**Review by rohanKanojia — APPROVED** [review]

**Comment by rohanKanojia** [comment]
Could you please rebase your branch against the latest master?

**Comment by jorsol** [comment]
This might be a little out of the scope of the issue, but, is possible to add an integration test that actually verifies that the conversion webhook is working?

**Comment by rohanKanojia** [comment]
@jorsol : Yes, it's a good idea. I had plans to test this out for a blogpost. Unfortunately, I'm very busy with Eclipse JKube gradle plugins and can't say that I'll be able to do it soon.

Would you like to do it?

**Comment by jorsol** [comment]
> @jorsol : Yes, it's a good idea. I had plans to test this out for a blogpost. Unfortunately, I'm very busy with Eclipse JKube gradle plugins and can't say that I'll be able to do it soon.
> 
> Would you like to do it?

I would be happy to collaborate, unfortunately, I'm very busy too, I was thinking that since @MUzairS15 is already working on this PR, probably could collaborate on that too. 😅

**Comment by rohanKanojia** [comment]
Umm, okay. It's upto him whether he wants to do it or not. I think this PR is good to merge and this can be tackled in a separate issue. I think we should create similar integration tests for testing admission webhooks as well

**Comment by MUzairS15** [comment]
@rohanKanojia @jorsol I would like to work for this, but I will not be able to do till 15-08-2021, as I have my end semester approaching.
I think as @rohanKanojia said separate issue can be  made and if anyone else is interested he/she may I try or I will.

**Pull request closed without merging**